packages/pluggableWidgets/notifications-native/src/Notifications.tsx

```diff
@@ -1,67 +1,139 @@
+            const actions = props.actions.filter(item => item.name === notification.data?.actionName);
 
-        if (props.guid) {
-            props.guid.setValue(data?.guid);
-        }
-        if (props.title) {
-            props.title.setValue(title);
-        }
-        if (props.subtitle) {
-            props.subtitle.setValue(subtitle);
-        }
-        if (props.body) {
-            props.body.setValue(body);
-        }
-        if (props.action) {
-            props.action.setValue(actions.join(" "));
-        }
+            if (actions.length === 0) {
+                return;
+            }
+            if (props.guid) {
+                props.guid.setValue(notification.data?.guid);
+            }
+            if (props.title) {
+                props.title.setValue(title);
+            }
+            if (props.subtitle) {
+                props.subtitle.setValue(subtitle);
+            }
+            if (props.body) {
+                props.body.setValue(body);
+            }
+            if (props.action) {
+                props.action.setValue(actions.map(action => action.name).join(" "));
+            }
 
-        actions.forEach(action => {
-            const handler = getHandler(action);
-            executeAction(handler);
-        });
-    };
+            actions.forEach(action => executeAction(getHandler(action)));
+        },
+        [props.actions, props.guid, props.title, props.subtitle, props.body, props.action]
+    );
 
-    const onReceive = (notification: FirebaseMessagingTypes.RemoteMessage): void => {
-        if (notification.notification) {
-            handleNotification(notification.notification, notification.data, action => action.onReceive);
-        }
-    };
+    const remoteMessageHandlerHelpers = (
+        notification: FirebaseMessagingTypes.Notification
+    ): Pick<Notification, "title" | "subTitle" | "body"> => ({
+        title: notification?.title,
+        body: notification?.body,
+        subTitle: notification.ios?.subtitle
+    });
+
+    const onReceive = useCallback(
+        (message: FirebaseMessagingTypes.RemoteMessage): void => {
+            const { notification } = message;
+            if (notification) {
+                handleNotification(
+                    {
+                        data: message.data,
+                        ...remoteMessageHandlerHelpers(notification)
+                    },
+                    action => action.onReceive
+                );
+            }
+        },
+        [handleNotification]
+    );
 
-    const onOpen = (notificationOpen: FirebaseMessagingTypes.RemoteMessage): void => {
-        if (notificationOpen.notification) {
-            handleNotification(notificationOpen.notification, notificationOpen.data, action => action.onOpen);
+    const onOpen = useCallback(
+        (message: FirebaseMessagingTypes.RemoteMessage): void => {
+            const { notification } = message;
+            if (notification) {
```

Comment:
Same comment here, also used message.data in line 105

Reply:
Ditto as above